pkg/driver/node_server.go

```diff
@@ -1964,6 +1965,36 @@ func (driver *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoReque
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	// Decode and check if the node is configured
+	nodeInfo, err := driver.flavor.GetNodeInfo(nodeID)
+	if err != nil {
+		log.Error("Cannot unmarshal node from node ID. err: ", err.Error())
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+
+	nodeLabels, err := driver.flavor.GetNodeLabelsByName(nodeInfo.Name)
+	log.Infof("node %s nodeLabels: %+v", nodeInfo.Name, nodeLabels)
+	if err != nil {
+		log.Error("Cannot load node labels from node. err: ", err.Error())
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+
+	accessableTopologySegments := make(map[string]string)
+	
+	for key, value := range nodeLabels {
+		log.Tracef("node %s label: %s => %s", nodeInfo.Name, key, value)
+		if key == defaultCSITopologyKey {
+			log.Infof("FOUND node %s label: %s => %s", nodeInfo.Name, key, value)
+			accessableTopologySegments[key] = value
+		}
+	}
+
+	topo := &csi.Topology{Segments: accessableTopologySegments}
+	if len(accessableTopologySegments) == 0 {
+		topo = nil
```

Comment:
Not sure if we have to keep this as nil or an empty csi.Topology object

Something like this,
var topo *csi.Topology
if len(accessableTopologySegments) > 0 {
    topo = &csi.Topology{Segments: accessableTopologySegments}
}